sdk/core/core-amqp/src/auth/sas.ts

```diff
@@ -78,9 +78,7 @@ export class SharedKeyCredential {
    * @param {string} connectionString - The EventHub/ServiceBus connection string
    */
   static fromConnectionString(connectionString: string): SharedKeyCredential {
-    const parsed = parseConnectionString<
-      ServiceBusConnectionStringModel & { SharedAccessSignature: string }
-    >(connectionString);
+    const parsed = parseConnectionString<ServiceBusConnectionStringModel>(connectionString);
```

Comment:
Should we be exposing `parseConnectionString` methods from service-bus and event-hubs separately?
Which would be 
```ts
export const parseSBConnectionString = (connectionString: string) =>
  parseConnectionString<ServiceBusConnectionStringModel>(connectionString);
```
This way, users would also get the type info.

Reply:
I would do that if the parseSBConnectionString method would then follow all the requirements logged in #11893 and #11894. If we are going with an intermediate offering in the meantime, then I would keep this as is in core-amqp

Also, this might be a good opportunity to gather all our core-amqp v2 requirements and see if we want to make the jump

Reply:
The other benefit of what @HarshaNalluru 's suggesting is that we can slim down the interface that we return since, as you've pointed out, the current interfaces have the `[x: string]: any;` which I'd rather not be on the hook for in the future.